datalad/support/s3.py

```diff
@@ -411,13 +414,15 @@ def get_versioned_url(url, guarantee_versioned=False, return_all=False, verify=F
         s3_bucket, fpath = fpath.split('/', 1)
     elif url_rec.scheme == 's3':
         s3_bucket = url_rec.hostname  # must be
-        # and for now implement magical conversion to URL
-        # TODO: wouldn't work if needs special permissions etc
-        # actually for now
-        raise NotImplementedError
+        if url_rec.query and 'versionId=' in url_rec.query:
+            was_versioned = True
+            all_versions.append(url)
```

Comment:
Good point! Indeed it should for all aws URLs. I will refactor tomorrow unless someone beats me to it by then

Reply:
I looked into this further.  We already dealt with this in 58f3d92f6 (BF: s3: Don't duplicate ID when getting versioned URL, 2018-04-05, #2384), so we should be covered for other URLs.  Sorry for the noise.

Reply:
Oh, but  it does point to inadequacy in comparison to http URLs (in the long run since s3 URLs support officially not implemented) when we request all versions?
Something to eat least add a comment about I think

Reply:
> but it does point to inadequacy in comparison to http URLs (in the long run since s3 URLs support officially not implemented) when we request all versions?

I'm not following what the inadequacy is, so I'll leave that comment to you.

Reply:
if I understood https://github.com/datalad/datalad/commit/58f3d92f6bbf3ffa915e28a4efaccedfcb828b56 correctly, if I provide a versioned http url, and request all versions -- it will get decomposed and I get all versions.  With current quick workaround I did, if I provide versioned s3 url, and request all versions -- only that version would be returned.

Reply:
OK, so you're talking about the same thing that you comment on in the first commit of this PR: the PR doesn't properly retrieve versions for `s3://` urls, but we'll pass through a `s3://` that already has a version tag.  Given that you said the initial question in this thread "pointed to inadequacy", I assumed you must have been referring to something else.